client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,137 @@
+
+/**
+ * Internal dependencies
+ */
+import shortcodeUtils from 'lib/shortcode';
+import { getVideo } from 'state/videos/selectors';
+import QueryVideo from 'components/data/query-video';
+
+class WpVideoView extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constrainVideoDimensions( shortcodeWidthAttribute, shortcodeHeightAttribute ) {
+		let { width, height } = this.props.video
+			? this.props.video
+			: {
+				width: 640,
+				height: 640 / ( 16 / 9 )
+			};
+		const aspectRatio = width / height;
+
+		if ( shortcodeWidthAttribute && ! shortcodeHeightAttribute ) {
+			width = shortcodeWidthAttribute;
+			height = shortcodeWidthAttribute / aspectRatio;
+		} else if ( ! shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			width = shortcodeHeightAttribute * aspectRatio;
+			height = shortcodeHeightAttribute;
+		} else if ( shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			width = shortcodeWidthAttribute;
+			height = shortcodeHeightAttribute;
+		}
+
+		return { width, height };
+	}
+
+	getVideoAttributes() {
+		const shortcode = shortcodeUtils.parse( this.props.content );
+		const namedAttrs = shortcode.attrs.named;
+
+		const videoDimensions = this.constrainVideoDimensions(
+			parseInt( namedAttrs.w, 10 ),
+			parseInt( namedAttrs.h, 10 )
+		);
+
+		const shortcodeAttributes = {
+			guid: shortcode.attrs.numeric[0],
+			w: videoDimensions.width,
+			h: videoDimensions.height,
+			autoplay: includes( [ 'true', '1' ], namedAttrs.autoplay ),
+			hd: includes( [ 'true', '1' ], namedAttrs.hd ),
+			loop: includes( [ 'true', '1' ], namedAttrs.loop ),
+			at: parseInt( namedAttrs.at, 10 ) || 0,
+			defaultLangCode: namedAttrs.defaultlangcode
+		};
+
+		return Object.assign(
+			pick( shortcodeAttributes, [ 'guid', 'w', 'h' ] ),
+			{ embedUrl: this.getEmbedUrl( shortcodeAttributes ) }
+		);
+	}
+
+	getEmbedUrl( shortcodeAttributes ) {
+		const defaultAttributeValues = { hd: false, at: 0, defaultLangCode: undefined };
+		const attributesWithNonDefaultValues = omitBy(
+			shortcodeAttributes,
+			( value, key ) => defaultAttributeValues[ key ] === value );
+		const queryStringAttributes = [ 'autoplay', 'hd', 'loop', 'at', 'defaultLangCode' ];
+		const queryString = QueryString.stringify( pick( attributesWithNonDefaultValues, queryStringAttributes ) );
+
+		return `https://videopress.com/embed/${ shortcodeAttributes.guid }?${ queryString }`;
+	}
+
+	onLoad() {
+		const doc = this.props.editor.iframeElement.contentDocument;
+		const script = doc.createElement( 'script' );
+		script.src = 'https://videopress.com/videopress-iframe.js';
+		script.type = 'text/javascript';
+		doc.getElementsByTagName( 'head' )[0].appendChild( script );
```

Comment:
This question had come up for me in the context of the media modal detail view, but do you think we should be smarter about whether we insert duplicate scripts? As it is, this can add up quickly when switching between editing modes:

![image](https://cloud.githubusercontent.com/assets/1779930/13263429/8818852c-da37-11e5-862f-60ab2ea0ad11.png)